### PR TITLE
Memoize POC

### PR DESCRIPTION
**Pull request by kresimir-coko**

### CODE REVIEW

**Review by bryceosterhaus — CHANGES_REQUESTED** [review]
@kresimir-coko can you write some tests for this as well to show it passes certain checks?


Ideally you should write some on your own from how you "expect" memoize to work. And then if you need further clarification, you can check [lodash's tests](https://github.com/lodash/lodash/blob/master/test/memoize.test.js) to see if you missed any edge cases

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/memoize.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+		const key = resolver ? resolver.apply(null, args) : args[0];
+
+		/**
+		 * Check if the first argument is cached, if yes, return it, otherwise store it
+		 */
+		if (cache.has(key)) {
+			return cache.get(key);
+		}
+		else {
+			const result = fn.apply(null, args);
```
Comment (bryceosterhaus) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/memoize.js:41]: Can you walk me through why you used `apply` here?

Comment (kresimir-coko) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/memoize.js:41]: We need to invoke the provided `fn` function somehow and accept arguments, so I figured that `apply` is the best at doing that.

This is the example usage which probably wasn't clear from the implementation:
```js
function randomFn(n) {return n};

const memoizedFn = memorize(randomFn);

memoizedFn(5);
```
